### PR TITLE
Hotfix/handling undefined user agent and async errors

### PR DESCRIPTION
**Pull request by ernestas-leliuga**

### CODE REVIEW

File: src/server/routes/partner-details.route.test.js [hunk 1]
```diff
@@ -666,7 +666,8 @@ describe("Partner Details Route: ", () => {
     });
 
     describe("When an error is thrown", () => {
```
Comment (simon-says-test) [inline src/server/routes/partner-details.route.test.js:668]: change describe text to make distinct from new test e.g. "When session.save throws an error"

File: src/server/routes/partner-details.route.js [hunk 2]
```diff
@@ -149,43 +149,52 @@ const partnerDetailsRouter = () => {
     });
   });
 
-  router.post("/continue", async (req, res) => {
+  router.post("/continue", async (req, res, next) => {
```
Comment (simon-says-test) [inline src/server/routes/partner-details.route.js:152]: this is good but I tihnk we need the same change for lines 32 and 112

File: src/server/routes/submit.route.test.js [hunk 3]
```diff
@@ -143,7 +143,8 @@ describe("Submit route: ", () => {
     });
 
     describe("When session.save throws an error", () => {
-      let response, res, req;
+      let response, res, req, next;
```
Comment (simon-says-test) [inline src/server/routes/submit.route.test.js:146]: response not required?

**Review by simon-says-test — COMMENTED** [review]
A few initial comments - nothing major

**Review by paul-burnett — COMMENTED** [review]
No further comments from me!

**Review by simon-says-test — APPROVED** [review]

**Pull request merged**